client/app/lib/providers/connectcompute.coffee

```diff
@@ -0,0 +1,88 @@
+React = require 'app/react'
+kd = require 'kd'
+shallowCompare = require 'react-addons-shallow-compare'
+
+# FIXME: so naive, but works for now.
+makeSingular = (plural) -> plural.slice 0, -1
+
+
+makeState = (config, props) ->
+
+  return {}  unless config.require
+
+  { storage } = kd.singletons.computeController
```

Comment:
To be honest, no. Do you have any suggestions on how we can do that? One thing came to my mind is that, we can first set a `loading` state, then once `computeController` is ready, we can set that loading state to `false`. a connected component can render its children depending on this `loading` state.

The tricky part is that, this needs to be a sync operation as much as possible, if we are gonna break that rule, we at least introduce a structured way of letting connected components decide on what they will render.

But maybe simply telling `mainView` to wait for `computeController` to be ready before rendering a view, might be a better approach.

I am just trying to think out loud, so i am open to any idea :)